packages/monaco/uplift-journal.md

```diff
@@ -0,0 +1,125 @@
+   > For preferences registered elsewhere in VSCode, look for references to the `editorConfigurationBaseNode` object defined in `src/vs/editor/common/config/editorConfigurationSchema.ts`
+
+With that system, what we get from Monaco will be in `packages/editor/src/browser/editor-generated-preference-schema.ts`, and anything we need to modify by hand will be in `packages/editor/src/browser/editor-preferences.ts`. Hopefully the latter will be quite a bit shorter, now :-).
+
+TODO: Make sure (i.e. write tests that prove) that our preference proxy can handle the de-flattening expected by VSCode. I think they may already exist...
+
+### New Errors
+
+I pulled master today, and it appears that plugins are working less well. Previously, I was getting good hovers etc. for TS, but today, the `Initializing JS/TS language features` spinner just spins. But eventually, it did stop, and then all plugin features, including peek views, appeared to be working.
+
+## 2/21/2022
+
+Working on the colorization-on-load bug. Definitely a problem with mime-type association timing. I also noticed as part of my debugging that there are various `-monaco-contribution.ts` files that register associations with specific files for e.g. `launch.json`, `tasks.json`. There is also a monkeypatch of the `Monaco.languages.register` function in the `monaco-frontend-application-contribution` that adds all registered languages to our list of potential language override identifiers. The former run before the latter, however, so those languages (`jsonc`, exclusively), don't get registered as language overrides - at least not that way. That may require implementing an override of the `PreferenceSchemaProvider` in the `monaco` package or `preferences` package (which depends on Monaco), so that we can get the correct behavior.
+
+### Observations re: colorization
+Language identification does fail on startup (all editors other than JSONC are incorrectly identified as `plaintext`), but events appear to be hooked up correctly so that editor languages are updated as plugins are processed, and colorization even fails for JSONC. It must not be a problem with mime-types, but something to do with colorization / tokenization itself.
+
+In the end, it turned out that we had just missed a change in the type of a particular field, and it wasn't nicely typed in the declaration file. I'm not sure how it wasn't throwing an error: apparently `Object.keys` always returns an empty array on a `Set`, so we were never trying to call methods that didn't exist; we just also weren't doing anything at all.
+
+### Setting definite version
+
+I pulled the 1.64.2 tag of VSCode today to pin my implementations to that. It is a couple of weeks behind what I'd pulled from `main`, and in n just that couple of weeks,a number of private API references were broken. We really shouldn't rely on private API.
```

Comment:
VS Code actually makes branches for their "monaco" releases, names sth.. like 0.32.x. What's your take on version names of what we publish for use in Theia?